File: Element.cpp
#include "Element.h"
#include "EngimonUniverse.h"

//4 sekawan
Element::Element(){
    this->setElementName("None");
}
Element::Element(string name){
    this->setElementName(name);
}
Element::Element(const Element& e){
    this->setElementName(e.elementName);
}
Element::~Element(){}

Element& Element::operator=(const Element& e){
    this->setElementName(e.elementName);
    return *this;
}
//setter
void Element::setElementName(string name){
    this->elementName = name;
}

//getter
string Element::getElementName(){
    return this->elementName;
}

//functions
int Element::elementToNumber(){
    if (this->getElementName()=="Fire"){
        return 0;
    }
    else if (this->getElementName()=="Water"){
        return 1;
    }
    else if (this->getElementName()=="Electric"){
        return 2;
    }
    else if (this->getElementName()=="Ground"){
        return 3;
    }
    else if (this->getElementName()=="Ice"){
        return 4;
    }
    return -1;
}

float Element::elementAdvantage(Element e){
    return elmtAdvTable[this->elementToNumber()][e.elementToNumber()];
}

File: java/Element.java
package willywangkyjava;

public class Element {
    final float[][] elmtAdvTable  = {{1,0,1,0.5f,2},{2,1,0,1,1},{1,2,1,0,1.5f},{1.5f,1,2,1,0},{0,1,0.5f,2,1}};
    String elementName;
    
    Element(){
        setElementName("None");
    }
    Element(String name){
        setElementName(name);
    }
    void setElementName(String name){
        this.elementName = name;
    }
    String getElementName(){
        return this.elementName;
    }
    int elementToNumber(){
        switch (this.elementName) {
            case "Fire":
                return 0;
            case "Water":
                return 1;
            case "Electric":
                return 2;
            case "Ground":
                return 3;
            case "Ice":
                return 4;
            default:
                return -1;
        }
    }
    float elementAdvantage(Element E){
        return elmtAdvTable[elementToNumber()][E.elementToNumber()];
    }
}
